From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lin Cheng <lin_cheng@dell.com>
Date: Mon, 17 May 2021 02:00:40 -0400
Subject: use openssl shared library replace boringssl

Thin OS smartcard will using openssl engines, boringssl not has engines
feature, so, we revert electron node module using back to openssl shared
library.

diff --git a/BUILD.gn b/BUILD.gn
index 99192976f2bdfd900240aa58b8989fe2c9d8c771..7fba955cbdf4a030794558c15a1b530ffd6e1f5c 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -138,11 +138,11 @@ config("node_internal_config") {
   if (node_module_version != "") {
     defines += [ "NODE_MODULE_VERSION=" + node_module_version ]
   }
-  if (is_component_build) {
+  #if (is_component_build) {
     defines += [
       "NODE_SHARED_MODE",
     ]
-  }
+  #}
 
   if (target_cpu == "x86") {
     node_arch = "ia32"
@@ -187,7 +187,7 @@ config("node_internal_config") {
   }
 }
 
-component("node_lib") {
+shared_library("node_lib") {
   deps = [
     ":node_js2c",
     "deps/cares",
@@ -259,7 +259,11 @@ component("node_lib") {
   }
 
   if (node_use_openssl) {
-    deps += [ "//third_party/boringssl" ]
+    libs += [
+      "ssl",
+      "crypto"
+    ]
+    #deps += [ "//third_party/boringssl" ]
     sources += [
       "src/node_crypto.cc",
       "src/node_crypto.h",
